src/dune_memory/dune_memory.ml

```diff
@@ -285,29 +313,107 @@ module Memory = struct
       Result.ok { root; build_root = None; repositories = []; handler }
 end
 
+let trimmable ?stats p =
+  let stats =
+    match stats with
+    | None -> Path.stat p
+    | Some stats -> stats
+  in
+  stats.Unix.st_nlink = 1
+
+let default_trim : trimming_result =
+  { trimmed_files = (0, []); trimmed_metafiles = [] }
+
+let _garbage_collect default_trim memory =
+  let path = Memory.path_meta memory in
+  let metas =
+    List.map ~f:(fun p -> (p, MetadataFile.parse p)) (FSSchemeImpl.list path)
+  in
+  let f default_trim = function
+    | p, Result.Error msg ->
+      Memory.with_lock memory (fun () ->
+          Log.infof "remove invalid metadata file %a: %s" Path.pp p msg;
+          Path.unlink_no_err p;
+          { default_trim with trimmed_metafiles = [ p ] })
+    | p, Result.Ok { MetadataFile.files; _ } ->
+      if
+        not
```

Comment:
Could you invert this if statement? This will put the shorter clause ahead and will get rid of the `not` to boot.